Firmware/pins_Rambo_1_3.h

```diff
@@ -18,85 +30,107 @@
+#define VOLT_PWR_PIN            4 //A4
+#define VOLT_BED_PIN            9 //A9
 
-#define LCD_PINS_RS            82
-#define LCD_PINS_ENABLE        18
-#define LCD_PINS_D4            19
-#define LCD_PINS_D5            70
-#define LCD_PINS_D6            85
-#define LCD_PINS_D7            71
 
 #define E0_STEP_PIN            34
 #define E0_DIR_PIN             43
 #define E0_ENABLE_PIN          26
 #define E0_MS1_PIN             65
 #define E0_MS2_PIN             66
-#define LED_PIN                13
 
-#ifdef THREEMM_PRINTER
-  #define FAN_PIN               8
-#else
-  #define FAN_PIN               6
-#endif
 
-#define KILL_PIN               -1 //80 with Smart Controller LCD
-#define SUICIDE_PIN            -1  //PIN that has to be turned on right after start, to keep power flowing.
-#define SDPOWER                -1
-#define HEATER_2_PIN           -1
+#define MOTOR_CURRENT_PWM_XY_PIN 46
+#define MOTOR_CURRENT_PWM_Z_PIN  45
+#define MOTOR_CURRENT_PWM_E_PIN  44
+#define SDPOWER             -1
+#define SDSS                53
+#define LED_PIN             13
+#define FAN_PIN              6
+#define FAN_1_PIN           -1
+#define PS_ON_PIN           -1
+#define KILL_PIN            -1  // 80 with Smart Controller LCD
+#define SUICIDE_PIN         -1  // PIN that has to be turned on right after start, to keep power flowing.
+
+#ifdef ULTRA_LCD
+
+//#define KILL_PIN            32
+
+#ifdef NEWPANEL
+
+#define BEEPER              84  // Beeper on AUX-4
+#define LCD_PINS_RS         82
+#define LCD_PINS_ENABLE     18
+#define LCD_PINS_D4	        19
+#define LCD_PINS_D5         70
+#define LCD_PINS_D6         85
+#define LCD_PINS_D7         71
+
+//buttons are directly attached using AUX-2
+#define BTN_EN1                72
+#define BTN_EN2                14
+#define BTN_ENC                 9  // the click
+
+#define SDCARDDETECT           15
+
 
-#define HEATER_0_PIN            3
-#define HEATER_BED_PIN          4
-#define FAN_1_PIN              -1 //6
-#define PS_ON_PIN              71
-#define MOTOR_CURRENT_PWM_XY_PIN   46
-#define MOTOR_CURRENT_PWM_Z_PIN    45
-#define MOTOR_CURRENT_PWM_E_PIN    44
+#endif //NEWPANEL
+#endif //ULTRA_LCD
+
+// Support for an 8 bit logic analyzer, for example the Saleae.
+// Channels 0-2 are fast, they could generate 2.667Mhz waveform with a software loop.
+#define LOGIC_ANALYZER_CH0		X_MIN_PIN		// PB6
+#define LOGIC_ANALYZER_CH1		Y_MIN_PIN		// PB5
+#define LOGIC_ANALYZER_CH2		53				// PB0 (PROC_nCS)
+// Channels 3-7 are slow, they could generate
+// 0.889Mhz waveform with a software loop and interrupt locking,
+// 1.333MHz waveform without interrupt locking.
+#define LOGIC_ANALYZER_CH3 		73				// PJ3
+// PK0 has no Arduino digital pin assigned, so we set it directly.
+#define WRITE_LOGIC_ANALYZER_CH4(value) if (value) PORTK |= (1 << 0); else PORTK &= ~(1 << 0) // PK0
+#define LOGIC_ANALYZER_CH5		16				// PH0 (RXD2)
+#define LOGIC_ANALYZER_CH6		17				// PH1 (TXD2)
+#define LOGIC_ANALYZER_CH7 		76				// PJ5
```

Comment:
Not connected on the miniRAMBo

Reply:
this pullrequest is just starting for MK2.5. Pinout is not final.